Sources/MountebankSwift/Models/Stub/Response/Proxy.swift

```diff
@@ -22,27 +22,43 @@ public struct Proxy: StubResponse, Codable, Equatable {
     /// Defines the replay behavior of the proxy.
     let mode: Mode
 
-    // TODO: add support for additional proxy options
-    // see https://github.com/MountebankSwift/MountebankSwift/issues/14
-    ///// An array of objects that defines how the predicates for new stubs are created.
-    ///// Each object in the array defines the fields to generate predicates from.
-    // let predicateGenerators: [PredicateGenerator]
-    ///// If true, mountebank will add a wait behavior to the response
-    ///// with the same latency that the proxied call took.
-    ///// This is useful in performance testing scenarios where you want to
-    ///// simulate the actual latency of downstream services that you're virtualizing.
-    // let addWaitBehavior: Bool
-    ///// If defined, mountebank will add a decorate behavior to the saved response.
-    // let addDecorateBehavior: String
-
-    public init(to: String, mode: Mode) {
+    /// An array of objects that defines how the predicates for new stubs are created.
+    /// Each object in the array defines the fields to generate predicates from.
+    let predicateGenerators: [PredicateGenerator]
+
+    /// If true, mountebank will add a wait behavior to the response
+    /// with the same latency that the proxied call took.
+    /// This is useful in performance testing scenarios where you want to
+    /// simulate the actual latency of downstream services that you're virtualizing.
+    let addWaitBehavior: Bool
+
+    /// If defined, mountebank will add a decorate behavior to the saved response
+    ///
+    /// Post-processes the response using JavaScript injection before sending it. You can use a decorate behavior to
+    /// add data to a proxied response or substitute data from the request into the response, for example. The value
+    /// passed into the decorate behavior is a JavaScript function that can take up to three values: the request,
+    /// the response, and a logger. You can either mutate the response passed in (and return nothing), or return an
+    /// altogether new response.
+    ///
+    /// The `--allowInjection` command line flag must be set to support this behavior.
+    let addDecorateBehavior: String?
+
+    public init(
+        to: String,
+        mode: Mode = .proxyOnce,
+        predicateGenerators: [PredicateGenerator] = [],
+        addWaitBehavior: Bool = false,
+        addDecorateBehavior: String? = nil
+    ) {
         self.to = to
         self.mode = mode
+        self.predicateGenerators = predicateGenerators
+        self.addWaitBehavior = addWaitBehavior
+        self.addDecorateBehavior = addDecorateBehavior
     }
 }
 
-// TODO: Implement and test Proxy Parameters
-// see https://github.com/MountebankSwift/MountebankSwift/issues/14
+// TODO: These are http/https only, what's the best place for them?
 struct ProxyParameters {
```

Comment:
Still not used, where shall we put these?

Reply:
Perhaps create an enum for the network specific parameters, just like we've done on Imposter.NetworkProtocol

```
enum ProxyParameters {
    struct HTTPProxyParameters {
        let cert: String?
        let key: String?

        ...
    }

    case http(HTTPProxyParameters)
}
```
And then give `Proxy` that as a property? 

```
public struct Proxy: StubResponse, Codable, Equatable {
    ...

    /// Additional proxy parameters
    let proxyParameters: ProxyParameters?

}
```

Reply:
Or aternatively, add a network protocol too

public struct Proxy: StubResponse, Codable, Equatable {
    ... 

    public enum NetworkProtocol: Codable, Equatable {
        case http(
            /// Key-value pairs of headers to inject into the proxied request.
            injectHeaders: [String: JSON] = [:]
        )
        case https(
            /// The SSL client certificate, a PEM-formatted string
            cert: String? = nil,
            /// The SSL client private key, a PEM-formatted string
            key: String? = nil,
            /// A valid cipher (see https://nodejs.org/api/tls.html#tls_modifying_the_default_tls_cipher_suite)
            /// For older (and insecure) https servers, this field allows you to override the cipher used to communicate
            ciphers: String? = nil,
            /// The SSL method to use
            /// A valid OpenSSL protocol method name
            secureProtocol: String? = nil,
            /// Shared passphrase used for a single private key
            passphrase: String? = nil,
            /// Key-value pairs of headers to inject into the proxied request.
            injectHeaders: [String: JSON]
        )

        case tcp(keepalive: Bool = false)
    }

    ... other properties

    /// Network protocol for this proxy with additional proxy parameters
    let networkProtocol: NetworkProtocol

    public init(
        ...
        networkProtocol: NetworkProtocol = .http()
    ) {

    }
}

I like this, but it might be less discoverable for users?

Reply:
I would opt for the second solution. To create a new struct like `NetworkProtocol` for http/https options.

Reply:
Ok, lets do that

Reply:
Changed